### PR TITLE
Add a way to disable/enable users

### PR DESCRIPTION
**Pull request by aledeg**
Changes proposed in this pull request:

- Add a way to disable/enable users

How to test the feature manually:

1. as an admin user, open the manage user page
2. check that all users are enabled (default value)
3. create a new user (it must be enabled)
4. logout and login with the newly created user
5. logout and login as an admin user
6. disable the newly created user
7. logout and try to login with the newly created user (it shouldn't work)

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested
- [ ] unit tests written (optional if too hard)
- [ ] documentation updated

[Additional information can be found in the documentation](https://github.com/FreshRSS/FreshRSS/tree/master/docs/en/developers/04_Pull_requests.md).

@Alkarex I've introduced the changes you've requested here → https://github.com/FreshRSS/FreshRSS/pull/2709#issuecomment-571283963

### CODE REVIEW

File: app/views/user/details.phtml [hunk 1]
```diff
@@ -68,11 +70,16 @@
...
                 <button type="submit" class="btn btn-attention confirm" name="action" value="purge"><?= _t('gen.action.purge') ?></button>
                 <button type="submit" class="btn btn-attention confirm" name="action" value="delete"><?= _t('gen.action.remove') ?></button>
-                <?php if ($isAdmin && !$this->isDefaultUser): ?>
+                <?php if ($isAdmin && !$isDefault): ?>
                     <button type="submit" class="btn btn-attention confirm" name="action" value="demote"><?= _t('gen.action.demote') ?></button>
                 <?php elseif (!$isAdmin): ?>
                     <button type="submit" class="btn btn-attention confirm" name="action" value="promote"><?= _t('gen.action.promote') ?></button>
                 <?php endif; ?>
+                <?php if ($enabled && !$isDefault): ?>
+                    <button type="submit" class="btn btn-attention confirm" name="action" value="disable"><?= _t('gen.action.disable') ?></button>
```
Comment (Alkarex) [inline app/views/user/details.phtml]:
```suggestion
                    <button type="submit" class="btn btn-attention" name="action" value="disable"><?= _t('gen.action.disable') ?></button>
```

image

This is a reversible action, so we might not need the confirmation dialogue box

File: app/views/user/details.phtml [hunk 2]
```diff
@@ -68,11 +70,16 @@
...
+                <?php elseif (!$enabled): ?>
+                    <button type="submit" class="btn btn-attention confirm" name="action" value="enable"><?= _t('gen.action.enable') ?></button>
```
Comment (Alkarex) [inline app/views/user/details.phtml]:
```suggestion
                    <button type="submit" class="btn btn-attention" name="action" value="enable"><?= _t('gen.action.enable') ?></button>
```

Comment (aledeg) [reply app/views/user/details.phtml]: @Alkarex done

**Comment by Alkarex** [comment]
We also need to disable users in:
* Actualize: https://github.com/FreshRSS/FreshRSS/blob/6edbeaaf6add5e6f60d949301a05ef42fbdc6afd/app/actualize_script.php#L56-L59
* API
* WebSub

**Comment by Alkarex** [comment]
Maybe also add this new information in the CLI `user-info.php`

**Comment by aledeg** [comment]
I'll look into that later!

**Comment by aledeg** [comment]
@Alkarex I've updated everything you've mentioned. I am not sure about my changes because it's the first time I read that code. Let me know if something is missing.
I feel that it's ready to be merged.

**Review by Alkarex — APPROVED** [review]

**Pull request merged**